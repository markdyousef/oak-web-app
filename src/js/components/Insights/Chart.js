import React, { PropTypes } from 'react';
import { PolarChart } from 'react-chartjs';

import css from './Insights.css';

const Chart = ({ items, type }: Object) => {
    return (
        <div className={css.chart}>
            {(() => {
                if(type === 'polar') {
                    return null;
                }
                return null;
            })()}
        </div>
    );
}

Chart.propTypes = {
    items: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        percentile: PropTypes.number.isRequired,
        children: PropTypes.arrayOf(PropTypes.object)
    })),
    type: PropTypes.string
}

Chart.defaultProps = {
    items: []
};

export default Chart;
