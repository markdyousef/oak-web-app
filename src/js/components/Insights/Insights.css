.container {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.insight {
    height: 250px;
    width: 250px;
}

.insight div {
    height: 100%;
    width: 100%;
}
