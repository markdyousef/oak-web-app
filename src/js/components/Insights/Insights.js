import React, { PropTypes, Component } from 'react';

import css from './Insights.css';

class Insights extends Component {
    static propTypes = {
        insights: PropTypes.shape({
            values: PropTypes.array
        })
    }
    static defaultProps = {
        insights: {}
    }
    constructor() {
        super();
        this.renderInsight = this.renderInsight.bind(this);
        this.state = {}
    }
    renderInsight(name: string) {
        const { insights } = this.props;

        if (insights[name]) {
            const items = insights[name];
        }
        return null;
    }
    render() {
        return (
            <div className={css.container}>
                {this.renderInsight('values')}
                {this.renderInsight('personality')}
                {this.renderInsight('needs')}
            </div>
        );
    }
}

export default Insights;
