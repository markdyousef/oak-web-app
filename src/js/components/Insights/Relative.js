import React, { PropTypes } from 'react';

import css from './Insights.css';

const Relative = ({ category, subCategories }: Object) => {
    return (
        <div className={css.relative}>

        </div>
    );
}

Relative.propTypes = {
    category: PropTypes.shape({
        name: PropTypes.string.isRequired,
        percentil: PropTypes.number.isRequired
    })
}

Relative.defaultProps = {
    category: {}
};

export default Relative;
