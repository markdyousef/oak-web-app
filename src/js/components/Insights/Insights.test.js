// @flow
import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Insights from './Insights';

test('renders Insights', () => {
    const props = {
        insights: {}
    };
    const shallowComponent = shallow(
        <Insights {...props} />
    );
    expect(shallowComponent).toMatchSnapshot();
});
