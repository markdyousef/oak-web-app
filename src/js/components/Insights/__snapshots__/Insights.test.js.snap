exports[`test renders Insights 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Insights
        insights={Object {}}
        insightsOther={Object {}} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Insights {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "insights": Object {},
          "insightsOther": Object {},
        },
        "refs": Object {},
        "renderInsight": [Function],
        "state": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}>
          
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Insights
    insights={Object {}}
    insightsOther={Object {}} />,
}
`;
